Warnings from Haskell Dockerfile Linter

The "Haskell Dockerfile Linter" Tool:
https://github.com/lukasmartinelli/hadolint
report the following issues:
https://github.com/koalaman/shellcheck/wiki/SC2155
https://github.com/koalaman/shellcheck/wiki/SC2086
for:
https://github.com/phpmyadmin/docker/blob/master/Dockerfile#L14-L28

(The MAINTAINER issue: https://github.com/lukasmartinelli/hadolint/wiki/DL4000 is false positive, more information in: https://github.com/phpmyadmin/docker/issues/85#issuecomment-261693890)

fix:
1. check the return value of (can be done by splitting export & assignment):
```
mktemp -d
```
2. double quote the strings that contain variables ($URL &  $VERSION) in
```
    && curl --output phpMyAdmin.tar.gz --location $URL \
    && curl --output phpMyAdmin.tar.gz.asc --location $URL.asc \
	&& mv phpMyAdmin-$VERSION-all-languages /www \
    && rm -rf /www/js/jquery/src/ /www/js/openlayers/src/ /www/setup/ /www/sql/ /www/examples/ /www/test/ /www/po/ /www/templates/test/ /www/phpunit.xml.* /www/build.xml  /www/composer.json /www/RELEASE-DATE-$VERSION \
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,7 +12,8 @@ ENV URL https://files.phpmyadmin.net/phpMyAdmin/${VERSION}/phpMyAdmin-${VERSION}
 
 # Download tarball, verify it using gpg and extract
 RUN set -x \
-    && export GNUPGHOME="$(mktemp -d)" \
+    && GNUPGHOME="$(mktemp -d)" \
+    && export GNUPGHOME \
     && apk add --no-cache curl gnupg \
     && curl --output phpMyAdmin.tar.gz --location $URL \
     && curl --output phpMyAdmin.tar.gz.asc --location $URL.asc \

